cranelift/codegen/src/isa/aarch64/inst/emit.rs

```diff
@@ -1337,6 +1337,31 @@ impl MachInstEmit for Inst {
                     }
                 }
             }
+            &Inst::MovPReg { rd, rm } => {
+                let rd = allocs.next_writable(rd);
+                let rm: Reg = rm.into();
```

Comment:
Can we assert that `rm` is one of a few registers on an allow-list?

Ideally the property we want here is that it is not an allocatable register. Moving from a register managed by the regalloc has undefined behavior: we don't know what the value will be. But grabbing the `MachineEnv` and doing an assert on the lists of registers there is sort of cumbersome, so it's probably better if we limit it first to e.g. `FP` and `SP` (with a comment describing the real condition) and then add more regs to the list if we need them.